### PR TITLE
Created a new component file and edited the logiin page

### PR DESCRIPTION
**Pull request by farisshomali**

### CODE REVIEW

File: lib/login.dart [hunk 1]
```diff
@@ -1,22 +1,245 @@
...
+import 'package:testingui/components/textfield.dart';
 import 'package:testingui/signup.dart';
 
-class Login extends StatelessWidget {
+class Login extends StatefulWidget {
+  @override
+  _LoginState createState() => _LoginState();
+}
+
+class _LoginState extends State<Login> {
```
Comment (7snalbar) [inline lib/login.dart:11]: Change to StatelessWidget since no State variables are being used

File: lib/login.dart [hunk 2]
```diff
@@ -1,22 +1,245 @@
...
-          Text("LogIn Page"),
-          MaterialButton(
-              child: Text("Go to Sign UP"),
-              onPressed: () {
-                Navigator.push(
-                    context, MaterialPageRoute(builder: (context) => SignUp()));
-              })
-        ],
+      backgroundColor: Colors.grey[100],
+      body: SafeArea(
```
Comment (7snalbar) [inline lib/login.dart:16]: SafeArea should be the parent of the Scaffold no the opposite

File: lib/login.dart [hunk 3]
```diff
@@ -1,22 +1,245 @@
...
+              child: Column(
+                children: <Widget>[
+                  Column(
+                    crossAxisAlignment: CrossAxisAlignment.center,
+                    children: <Widget>[
+                      Stack(
+                        children: <Widget>[
+                          Image.network("https://i.imgur.com/EjmAHKZ.png"),
+                          Padding(
+                            padding: const EdgeInsets.only(top: 355),
```
Comment (7snalbar) [inline lib/login.dart:30]: Stack should be given Alignment.bottomCenter instead of Padding

File: lib/login.dart [hunk 4]
```diff
@@ -1,22 +1,245 @@
...
+                                borderSide: BorderSide(color: Colors.blue),
+                                borderRadius:
+                                    BorderRadius.all(Radius.circular(30.0))),
+                            prefixIcon: Icon(CupertinoIcons.person),
+                            hintText: "Email",
+                            fillColor: Colors.white,
+                            filled: true,
+                          ),
+                        ),
+                      ),
```
Comment (7snalbar) [inline lib/login.dart:73]: use a reusable textfield where it is only given

1. Icon
2. hintText

File: lib/login.dart [hunk 5]
```diff
@@ -1,22 +1,245 @@
...
+                                borderSide: BorderSide(color: Colors.blue),
+                                borderRadius:
+                                    BorderRadius.all(Radius.circular(30.0))),
+                            prefixIcon: Icon(CupertinoIcons.padlock),
+                            hintText: "Password",
+                            fillColor: Colors.white,
+                            filled: true,
+                          ),
+                        ),
+                      ),
```
Comment (7snalbar) [inline lib/login.dart:97]: use a reusable textfield where it is only given

Icon
hintText

File: lib/login.dart [hunk 6]
```diff
@@ -1,22 +1,245 @@
...
+                              color: Colors.blue[700],
+                              onPressed: () {},
+                              child: Text(
+                                "LOG IN",
+                                style: TextStyle(
+                                    color: Colors.white,
+                                    fontWeight: FontWeight.bold),
+                              )),
+                        ),
+                      ),
```
Comment (7snalbar) [inline lib/login.dart:126]: use reusable button where it is only given 

1. width
2. height
3. onPressed function
4. Text

File: lib/login.dart [hunk 7]
```diff
@@ -1,22 +1,245 @@
...
+                            ],
+                          ),
+                        ),
+                      ),
+                    ],
+                  ),
+                ],
+              ),
+            ),
+          ),
```
Comment (7snalbar) [inline lib/login.dart:241]: Use reusable social login button by only passing the following:
1. Icon url
2. text
3. color
4. width 
5. height

File: lib/login.dart (hunk 3 above)
Comment (farisshomali) [reply lib/login.dart:30]: This Will not Work,

File: lib/login.dart (hunk 1 above)
Comment (farisshomali) [reply lib/login.dart:11]: Done

File: lib/login.dart (hunk 2 above)
Comment (farisshomali) [reply lib/login.dart:16]: Done

File: lib/login.dart (hunk 4 above)
Comment (farisshomali) [reply lib/login.dart:73]: Done

File: lib/login.dart (hunk 5 above)
Comment (farisshomali) [reply lib/login.dart:97]: Done

**Pull request merged**